pkg/agent/utils.go

```diff
@@ -447,25 +472,113 @@ func getDynamicKubeletConfigFileContent(kc map[string]string) string {
 	kubeletConfig.SystemReserved = strKeyValToMap(kc["--system-reserved"], ",", "=")
 	kubeletConfig.KubeReserved = strKeyValToMap(kc["--kube-reserved"], ",", "=")
 
+	// settings from customKubeletConfig, only take if it's set
+	if customKc != nil {
+		if customKc.CPUManagerPolicy != "" {
+			kubeletConfig.CPUManagerPolicy = customKc.CPUManagerPolicy
+		}
+		if customKc.CPUCfsQuota != nil {
+			kubeletConfig.CPUCFSQuota = customKc.CPUCfsQuota
+		}
+		if customKc.CPUCfsQuotaPeriod != "" {
+			kubeletConfig.CPUCFSQuotaPeriod = Duration(customKc.CPUCfsQuotaPeriod)
+		}
+		if customKc.TopologyManagerPolicy != "" {
+			kubeletConfig.TopologyManagerPolicy = customKc.TopologyManagerPolicy
+		}
+		if customKc.ImageGcHighThreshold != nil {
+			kubeletConfig.ImageGCHighThresholdPercent = customKc.ImageGcHighThreshold
+		}
+		if customKc.ImageGcLowThreshold != nil {
+			kubeletConfig.ImageGCLowThresholdPercent = customKc.ImageGcLowThreshold
+		}
+		if customKc.AllowedUnsafeSysctls != nil {
+			kubeletConfig.AllowedUnsafeSysctls = customKc.AllowedUnsafeSysctls
+		}
+	}
+
 	configStringByte, _ := json.MarshalIndent(kubeletConfig, "", "    ")
 	return string(configStringByte)
 }
 
+// GetSysctlConfigFileContent gets sysctl config file content
+// this func sorts the keys so that the order is fixed (mainly for test assertions)
+func GetSysctlConfigFileContent(osConfig *datamodel.CustomOSConfig) string {
```

Comment:
can we move this to template as we get WYSIWYG

Reply:
moved to template